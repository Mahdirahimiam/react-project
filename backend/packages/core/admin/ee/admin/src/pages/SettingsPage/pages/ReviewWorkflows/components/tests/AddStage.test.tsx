import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { render } from '@testing-library/react';

import { AddStage } from '../AddStage';

const ComponentFixture = () => (
  <ThemeProvider theme={lightTheme}>
    <AddStage>Add stage</AddStage>
  </ThemeProvider>
);

const setup = () => render(<ComponentFixture />);

describe('Admin | Settings | Review Workflow | AddStage', () => {
  it('should render a list of stages', () => {
    const { container, getByText } = setup();

    expect(container).toMatchSnapshot();
    expect(getByText('Add stage')).toBeInTheDocument();
  });
});
