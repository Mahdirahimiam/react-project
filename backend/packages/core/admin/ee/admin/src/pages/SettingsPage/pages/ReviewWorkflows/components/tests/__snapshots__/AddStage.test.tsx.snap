// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin | Settings | Review Workflow | AddStage should render a list of stages 1`] = `
.c7 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c6 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #8e8ea9;
}

.c0 {
  background: #ffffff;
  padding-top: 12px;
  padding-right: 16px;
  padding-bottom: 12px;
  padding-left: 16px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c4 > circle {
  fill: #eaeaef;
}

.c4 > path {
  fill: #666687;
}

.c1 {
  border-radius: 26px;
}

.c1 svg {
  height: 24px;
  width: 24px;
}

.c1 svg > path {
  fill: #666687;
}

.c1:hover {
  color: #4945ff !important;
}

.c1:hover .c5 {
  color: #4945ff !important;
}

.c1:hover .c3 > circle {
  fill: #4945ff;
}

.c1:hover .c3 > path {
  fill: #f6f6f9;
}

.c1:active .c5 {
  color: #4945ff;
}

.c1:active .c3 > circle {
  fill: #4945ff;
}

.c1:active .c3 > path {
  fill: #f6f6f9;
}

<div>
  <button
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <svg
        aria-hidden="true"
        class="c3 c4"
        fill="none"
        height="1rem"
        viewBox="0 0 24 24"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          fill="#212134"
          r="12"
        />
        <path
          d="M17 12.569c0 .124-.1.224-.225.224h-3.981v3.982c0 .124-.101.225-.226.225h-1.136a.225.225 0 0 1-.226-.225v-3.981H7.226A.225.225 0 0 1 7 12.567v-1.136c0-.125.1-.226.225-.226h3.982V7.226c0-.124.1-.225.224-.225h1.138c.124 0 .224.1.224.225v3.982h3.982c.124 0 .225.1.225.224v1.138Z"
          fill="#F6F6F9"
        />
      </svg>
      <span
        class="c5 c6"
      >
        Add stage
      </span>
    </div>
  </button>
  <div
    class="c7"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
